libreoffice-core/vcl/source/outdev/font.cxx

```diff
@@ -384,6 +387,49 @@ void OutputDevice::BeginFontSubstitution()
     pSVData->maGDIData.mbFontSubChanged = false;
 }
 
+void OutputDevice::AddCustomMacroFonts()
+{
+    OutputDevice* pDevice = Application::GetDefaultDevice();
+    pDevice->ImplAddCustomMacroFonts();
+}
+
+void OutputDevice::ImplAddCustomMacroFonts()
+{
+    std::list<std::string> fontNames = {
+        "Carlito",
+        "Carlito-Bold",
+        "Carlito-Italic",
+        "Carlito-BoldItalic",
+        "Caladea",
+        "Caladea-Bold",
+        "Caladea-Italic",
+        "Caladea-BoldItalic"
+    };
+
+    OUString installationRootPath(
+        getOfficePath(psp::whichOfficePath::InstallationRootPath)
+    );
+    std::string extension = ".ttf";
+
+    OUStringBuffer basePathBuffer(256);
+    basePathBuffer.appendAscii("file://");
+    basePathBuffer.append(installationRootPath);
+    basePathBuffer.appendAscii("/");
+    basePathBuffer.append(LIBO_SHARE_RESOURCE_FOLDER);
+    basePathBuffer.append("/macro_fonts/");
+    OUString basePath = basePathBuffer.makeStringAndClear();
+
+    for (std::string name : fontNames) {
+        OUStringBuffer fontPathBuffer(256);
+        fontPathBuffer.append(basePath);
+        fontPathBuffer.appendAscii(name.data());
+        fontPathBuffer.appendAscii(extension.data());
+        OUString filePath = fontPathBuffer.makeStringAndClear();
+        OUString fontName = OUString::createFromAscii(name);
+        OutputDevice::AddTempDevFont(filePath, fontName);
+    }
```

Comment:
For loops over strings, especially literals, access by reference (`auto&`) not by copy (`auto`). Shouldn't need OUStringBuffer for cases like this.

```suggestion
    for (auto& name : fontNames) {
        OutputDevice::AddTempDevFont(basePath + name + u".ttf", fontName);
    }
```

Reply:
Much tidier!